src/main/com/fulcrologic/rad/report.cljc

```diff
@@ -376,6 +376,27 @@
         (start-report! app report-class {:route-params route-params})
         (comp/transact! app [(dr/target-ready {:target report-ident})])))))
 
+(s/def ::arglist (s/and vector?
```

Comment:
The error message from this will be total crap. This should be an analysis inside of the macro

Reply:
I agree that e.g.

```
-- Spec failed --------------------

  (... [this this] ...)
       ^^^^^^^^^^^

should satisfy

  (fn
   [[this props]]
   (and
    (not= this props)
    (not= this '_)
    (not= props '_)))
```

is very far from the best error message ever but it still seems a big improvement over the current error message?